crates/ruff_server/src/fix.rs

```diff
@@ -165,3 +165,44 @@ pub(crate) fn fix_all(
         .collect())
     }
 }
+
+pub(crate) fn parse_all(query: &DocumentQuery, fixes: Fixes) -> DocumentQuery {
```

Comment:
feel free to rewrite this function. I think it could be better implementation at here.